packages/extensions/src/select-block/select-block-keymap.spec.ts

```diff
@@ -0,0 +1,197 @@
+    tr.setSelection(TextSelection.create(tr.doc, 2, 13))
+    editor.view.dispatch(tr)
+
+    await pressKey('mod-a')
+
+    const selection = editor.state.selection
+    expect(selection).toBeInstanceOf(TextSelection)
+    // Should select the entire blockquote, not just the paragraph
+    expect(selection.from).toBe(1) // Start of blockquote
+    expect(selection.to).toBe(14) // End of blockquote
+  })
+
+  it('should select list container when text block is fully selected', async () => {
+    const { editor, n } = setupTestWithSelectBlock()
+
+    const doc = n.doc(n.list({ kind: 'bullet' }, n.paragraph('Item 1'), n.paragraph('Item 2')))
+    editor.set(doc)
+    // Select all text in first paragraph
+    const tr = editor.state.tr
+    tr.setSelection(TextSelection.create(tr.doc, 2, 9))
+    editor.view.dispatch(tr)
+
+    await pressKey('mod-a')
+
+    const selection = editor.state.selection
+    expect(selection).toBeInstanceOf(TextSelection)
+    // Should select the entire list, not just the paragraph
+    expect(selection.from).toBe(1) // Start of list
+    expect(selection.to).toBeGreaterThan(9) // End of list
+  })
+
+  it('should select all blocks when selection spans multiple blocks', async () => {
+    const { editor, n } = setupTestWithSelectBlock()
+
+    const doc = n.doc(n.paragraph('First'), n.paragraph('Second'), n.paragraph('Third'))
+    editor.set(doc)
+    // Select across multiple paragraphs
+    const tr = editor.state.tr
+    tr.setSelection(TextSelection.create(tr.doc, 2, 15))
+    editor.view.dispatch(tr)
+
+    await pressKey('mod-a')
+
+    const selection = editor.state.selection
+    expect(selection).toBeInstanceOf(TextSelection)
+    // Should select from first block start to last block end
+    expect(selection.from).toBe(1) // Start of first paragraph
+    expect(selection.to).toBeGreaterThan(15) // End of last paragraph
+  })
+
+  it('should select heading block', async () => {
+    const { editor, n } = setupTestWithSelectBlock()
+
+    const doc = n.doc(n.heading({ level: 1 }, 'Hello <a>world'))
+    editor.set(doc)
+
+    await pressKey('mod-a')
+
+    const selection = editor.state.selection
+    expect(selection).toBeInstanceOf(TextSelection)
+    expect(selection.from).toBe(1)
+    expect(selection.to).toBe(12)
+  })
+
+  it('should do nothing if already all selected', async () => {
+    const { editor, n } = setupTestWithSelectBlock()
+
+    const doc = n.doc(n.paragraph('Hello world'))
+    editor.set(doc)
+    const tr = editor.state.tr
+    tr.setSelection(new AllSelection(tr.doc))
+    editor.view.dispatch(tr)
+
+    await pressKey('mod-a')
+
+    const selection = editor.state.selection
+    expect(selection).toBeInstanceOf(AllSelection)
+  })
+
+  it('should use selectBlock command directly', () => {
+    const { editor, n } = setupTestWithSelectBlock()
+
+    const doc = n.doc(n.paragraph('Hello <a>world'))
+    editor.set(doc)
+
+    const result = editor.commands.selectBlock?.()
```

Comment:
I guess we don't need this ? here

Reply:
I don't quite understand the meaning of this comment.Don't you mean that we don't need to test the command selectBlock separately?